src/lib.rs

```diff
@@ -113,11 +114,11 @@ impl<'a, 'b> plugin::Plugin<Request<'a, 'b>> for Json {
 pub struct Struct<T: Decodable> {
     marker: marker::PhantomData<T>
 }
-impl<T: 'static + Decodable> Key for Struct<T> {
+impl<T: 'static + Decodable + marker::Reflect> Key for Struct<T> {
     type Value = Option<T>;
 }
 
-impl<'a, 'b, T: 'static + Decodable> plugin::Plugin<Request<'a, 'b>> for Struct<T> {
+impl<'a, 'b, T: 'static + Decodable + marker::Reflect> plugin::Plugin<Request<'a, 'b>> for Struct<T> {
```

Comment:
Instead of using `Reflect`, you can replace `'static` and `Reflect` with an `Any` bound. This avoids the unstable API since `Any` is `#[stable]`.